### DISCUSSION TITLE
Reservation 관련 논의

### DISCUSSION THREAD

**Discussion by JuneParkCode** [manito42/FrontEnd#94] [Q&A]
## Reservation 카드 홀더에 대한 논의
![image](https://github.com/manito42/FrontEnd/assets/81505228/7f97791b-7070-486c-9601-05a24e2ff9fe)

- 위 디자인을 보시면, 홈의 상단에 Reservation 카드들이 위치한걸 보실 수 있습니다.
- 이 카드 홀더에 담고있는 데이터에 대해서 논의가 필요해보입니다.
- @falconlee236 님과의 대화에서는 "요청" 에 대한 카드들만 위치하자는 이야기를 했습니다
  - 멘티 입장
    - 본인이 요청한 멘토링 예약 (아직 수락안됨)
    - 멘토가 수락했으나, 본인이 확인하지 않은 멘토링 (확인 대기)
  - 멘토 입장
    - 멘토링 요청 (미확인)
    - 멘토링 요청 (확인했으나, 멘티가 수락한걸 확인하지 않은 경우)
  - 참고로, 모든 멘토링 예약 관련 변경 사항은 멘토 혹은 멘티에게 이메일로 알림이 갑니다.

### 논의가 필요한 사항
1. 카드 홀더에 위치하는 카드들의 상태에 대한 논의가 필요합니다.
  - Active (요청, 수락, 멘토링 피드백 대기) 에 대한 카드만 보여줄지
  - Request  (요청, 수락, 확인 대기) 상태에 대한 카드만 보여줄지
    - Mentoring Reservation 흐름이 살짝 바뀝니다.
2. 이에 앞서, 이 카드 홀더가 Home에 반드시 필요한가? 라는 것도 확실하게 해야할 것 같습니다.
  - 해당 카드들이 Reservation Page 에서는 필요한 것은 맞으나, Home 에서 필요한 정보인가에 대해서 조금 생각을 해봐야할 것 같습니다.

각자 생각을 남겨주시면 감사하겠습니다.

@falconlee236 @koreanddinghwan @9utty @obvoso @jinaji

**Comment by koreanddinghwan** [comment]
[현재 멘토링 예약 프로세스 보기](https://github.com/manito42/backend/wiki/Reservation#%EB%A9%98%ED%86%A0%EB%A7%81-%EC%98%88%EC%95%BD)

**Comment by koreanddinghwan** [comment]
2를 먼저 답변하자면/ 현재 저희 핵심 비즈니스가 멘토링이다보니 가장 핵심적인 서비스인 멘토링 예약을 home에 둔 설계 아닌가 싶습니다.

**Comment by obvoso** [comment]
1. 메인 페이지에 reservation 스크롤 컴포넌트를 배치할거면, 중요한 정보를 담고있어야 한다고 생각해요
 active와 request의 차이가 각각 멘토링 피드백 대기, 확인 대기의 유무인 거 같은데,
 둘 중 후자가 사용자에게 더 중요한 정보인 거 같아요
 
 2. 카드 홀더가 메인에 없으면 신규 사용자에게는 멘토링 서비스가 직관적일 수 있겠지만
 이미 서비스를 이용하는 사용자 입장에서는 카드 홀더를 배치해주는게 좋다고 생각합니다..

**Reply by koreanddinghwan** [reply]
ㅇㅇ 카드홀더 있으면 렌더링하고, 없으면 렌더링 안하던가(컴포넌트를)가 가능하지 않나요?(잘 모름, 아마 될 것 같음, 우리에겐 조건문이 있잖아)

**Comment by jinaji** [comment]
1. 피드백 대기보다는 확인 대기에 대한 정보를 사용자들이 더 궁금해 할 것 같습니다 둘 중 하나만 넣어야 한다면 request 쪽이 낫지 않을까 해여
2. 구현된 서비스 안에서는 멘토링 서비스가 주인만큼 접근성이 가장 좋은 홈에 위치시키는 거 좋다고 생각해여 ~ 
위에서 말씀해주신 것처럼 조건부 렌더링 하면 되지 않을까용

**Comment by falconlee236** [comment]
1. 저도 피드백은 딱히 필요 없을것 같습니다. 또한 피드백이 있으면 홈 화면이 지저분해 보일것 같아요
2. 사용자가 이 서비스를 사용하는 이유는 멘토링 예약현황 확인인데, 그 내용이 홈에 있는 것이 좋을 것 같습니다!! 즉 카드가 있는게 좋을것 같아요

**Comment by koreanddinghwan** [comment]
취업하셔서 (뜻하지 않은?) 게임개발중이신 구리님 빼곤 모두 1번에 대해 확인 대기가 있어야한다고 말씀해주셨네요, 
예약 프로세스가 변동되는 사안이라 

- [현재 멘토링 예약 프로세스 보기](https://github.com/manito42/backend/wiki/Reservation#%EB%A9%98%ED%86%A0%EB%A7%81-%EC%98%88%EC%95%BD) 여기서 설계 변경하고
- [[Fix] Reservation 프로세스에 CHECKED 추가 #66](https://github.com/manito42/backend/issues/66)에서 작업하도록 하면될까요?

**Reply by JuneParkCode** [reply]
네 예약 프로세스 그렇게 바꾸도록합시다.
추가로, Home 에서 보여주는 reservation 들에 대해서 조금 바꿔야하겠네요. user/id/reservation  이부분이 변경되어야합니다. 나머지는 백엔드 이슈에서 자세히 다루도록합시다.

**Comment by JuneParkCode** [comment]
홈에는 요청 관련된 카드만 보여주는 것으로 하겠습니다~ 땅땅땅
